### PR TITLE
 tạo biểu đồ requests được chấp nhận theo tuần, tháng

### PR DESCRIPTION
**Pull request by nguyenvanthanh1906**
## Related Tickets:
- [#xxxxx](https://edu-redmine.sun-asterisk.vn/issues/xxxxx)

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by nguyenvanthanh1906** [comment]
ready

**Comment by nguyenvanthanh1906** [comment]
ready

File: app/Http/Controllers/ChartsController.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+       
+        return view('admin.charts.index');
+    }
+
+    public function week($week) 
+    {
+        $requests = CRequest::where('is_approve', 1)->get();
+        $requests_of_week = [];
+        foreach($requests as $request)
+        {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ChartsController.php]:
```suggestion
        foreach($requests as $request) {
```

File: app/Http/Controllers/ChartsController.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     */
+    public function index()
+    {
+       
+        return view('admin.charts.index');
+    }
+
+    public function week($week) 
+    {
+        $requests = CRequest::where('is_approve', 1)->get();
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ChartsController.php]: 1 là gì? đưa vào config hoặc constant

File: app/Http/Controllers/ChartsController.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        $requests_of_week = [];
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ChartsController.php]:
```suggestion
        $requestsOfWeek = [];
```

File: app/Http/Controllers/ChartsController.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            }
+        }
+        $number_requests_week = [];
+        for ( $i = 1; $i <= 7; $i++)
+        {
+            $number_requests_week[] = 0;
+        }
+
+        foreach($requests_of_week as $request)
+        {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ChartsController.php]: convention

File: public/js/chart.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+var ctx = document.getElementById('myChart').getContext('2d');
+var url = 'charts/week/'+$('#weeks').val()
+var myChart = new Chart(ctx, {
+    type: 'bar',
+    data: {
+        labels: ['Monday', 'Tuesday', 'Wedsday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
+        datasets: [{
+            data: [],
+            label: 'Week',
+            backgroundColor: [
```
Comment (huyenttm-0901) [inline public/js/chart.js:11]: các option này có ý nghĩa gì?

File: public/js/chart.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                'rgba(255, 114, 0)'
+            ],
+            
+        }]
+    },
+    options: {
+        scales: {
+            y: {
+                beginAtZero: true
+            }
```
Comment (huyenttm-0901) [inline public/js/chart.js:27]: option này dùng để làm gì?

File: public/js/chart.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    }
+});
+document.getElementById("weeks").onchange = function() {
+    let url = 'charts/week/'+$('#weeks').val()
+    console.log(url)
+    $.ajax({
+        async:false,
+        url:url,
+        method:"GET", 
+        data:{
```
Comment (huyenttm-0901) [inline public/js/chart.js]:
```suggestion
        async: false,
        url: url,
        method: "GET", 
        data: { 
```

**Review by huyenttm-0901 — COMMENTED** [review]
2 commits
chú ý convention

File: app/Http/Controllers/ChartsController.php (hunk 2 above)
Comment (nguyenvanthanh1906) [reply app/Http/Controllers/ChartsController.php]: dạ 1 là true đó chị

File: public/js/chart.js (hunk 5 above)
Comment (nguyenvanthanh1906) [reply public/js/chart.js:11]: data : 1 array dữ liệu cho chart
label : tạo 1 nút dùng để ẩn/hiện chart
backgroundColor: màu nền của các cột trong chart
labels : tên các cột trong chart

File: public/js/chart.js (hunk 6 above)
Comment (nguyenvanthanh1906) [reply public/js/chart.js:27]: để cột y bắt đầu từ 0 chị

**Comment by nguyenvanthanh1906** [comment]
> 2 commits
> chú ý convention

dạ do bị conflic nên mới thành 2 commit đó chị

**Comment by nguyenvanthanh1906** [comment]
ready

File: app/Http/Controllers/ChartsController.php [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    {
+       
+        return view('admin.charts.index');
+    }
+
+    public function week($week) 
+    {
+        $requests = CRequest::where('is_approve', true)->get();
+        $requestsOfWeek = [];
+        foreach($requests as $request) {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ChartsController.php]:
```suggestion
        foreach ($requests as $request) {
```
sửa hết mấy chỗ tương tự

File: public/js/chart.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        data: { 
+            _token:_token,
+        },
+        success:function(data){ 
+            myChartMonth.data.labels = data['daysOfMonth']
+            myChartMonth.data.datasets[0].data = data['requestsOfMonth']
+            myChartMonth.update()
+        }
+    });
+};
```
Comment (huyenttm-0901) [inline public/js/chart.js]: eof

File: public/js/chart.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    success:function(data){ 
+        myChartMonth.data.labels = data['daysOfMonth']
+        myChartMonth.data.datasets[0].data = data['requestsOfMonth']
+        myChartMonth.update()
+    }
+});
+
+document.getElementById("months").onchange = function() {
+    let url = 'charts/month/'+$('#months').val()
+    console.log(url)
```
Comment (huyenttm-0901) [inline public/js/chart.js]: bỏ console đi

File: public/js/chart.js [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    success:function(data){ 
+        myChartMonth.data.labels = data['daysOfMonth']
+        myChartMonth.data.datasets[0].data = data['requestsOfMonth']
+        myChartMonth.update()
+    }
+});
+
+document.getElementById("months").onchange = function() {
+    let url = 'charts/month/'+$('#months').val()
```
Comment (huyenttm-0901) [inline public/js/chart.js]:
```suggestion
    let url = 'charts/month/' + $('#months').val()
```

File: public/js/chart.js [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+        _token:_token,
+    },
+    success:function(data){ 
+        myChartMonth.data.labels = data['daysOfMonth']
+        myChartMonth.data.datasets[0].data = data['requestsOfMonth']
+        myChartMonth.update()
+    }
+});
+
+document.getElementById("months").onchange = function() {
```
Comment (huyenttm-0901) [inline public/js/chart.js]: sao e ko dùng jquery?

File: resources/views/admin/charts/index.blade.php [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+<div class="container">
+    <center><h1>{{trans('requests.weeklystatistics')}}</h1></center>
+    <label for="weeks">{{trans('requests.week')}}</label>
+    <select name="weeks" class="form-control w-25" id="weeks">
+        <option value="{{date('W')}}" >{{date('W')}}</option>
+        <option value="{{date('W')-1}}" >{{date('W')-1}}</option>
+        <option value="{{date('W')-2}}" >{{date('W')-2}}</option>
+        <option value="{{date('W')-3}}" >{{date('W')-3}}</option>
+        <option value="{{date('W')-4}}" >{{date('W')-4}}</option>
+        <option value="{{date('W')-5}}" >{{date('W')-5}}</option>
```
Comment (huyenttm-0901) [inline resources/views/admin/charts/index.blade.php]: convention

File: resources/views/admin/charts/index.blade.php [hunk 14]
```diff
@@ -0,0 +1,29 @@
+@extends('admin.layouts.app', ['page' => 'home'])
+@section('content')
+<div class="container">
+    <center><h1>{{trans('requests.weeklystatistics')}}</h1></center>
+    <label for="weeks">{{trans('requests.week')}}</label>
+    <select name="weeks" class="form-control w-25" id="weeks">
+        <option value="{{date('W')}}" >{{date('W')}}</option>
+        <option value="{{date('W')-1}}" >{{date('W')-1}}</option>
+        <option value="{{date('W')-2}}" >{{date('W')-2}}</option>
```
Comment (huyenttm-0901) [inline resources/views/admin/charts/index.blade.php]: sao ko dùng for? mà sao lại là từ 0 - 5 thế em?

File: resources/views/admin/layouts/app.blade.php [hunk 15]
```diff
@@ -22,6 +22,7 @@
     <script src="{{asset('js/ajax.js')}}"></script>
     <script src="{{asset('js/pusher.js')}}"></script>
     <script src="{{asset('js/select2.js')}}"></script>
+    <script src="{{asset('js/chart.min.js')}}"></script>
```
Comment (huyenttm-0901) [inline resources/views/admin/layouts/app.blade.php]:
```suggestion
    <script src="{{ asset('js/chart.min.js') }}"></script>
```

**Comment by nguyenvanthanh1906** [comment]
ready
